rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,417 @@
+
+                if swap_amount and sell_amount:
+                    break
+        else:
+            log.error(
+                f'Failed to decode fee from swap token to token '
+                f'for transaction {transaction.tx_hash.hex()} due to sell_amount '
+                'or swap_amount being null.',
+            )
+            return
+
+        fee_event = self._create_fee_event(
+            tx_hash=transaction.tx_hash,
+            timestamp=transaction.timestamp,
+            fee_raw=sell_amount - swap_amount,
+            fee_asset=out_event.asset.resolve_to_evm_token(),
+            sender=sender,
+        )
+        decoded_events.append(fee_event)
+
+        return
+
+    def _process_swap(self,
+            transaction: 'EvmTransaction',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        """
+        Process a swap transaction by decoding the relevant events and logs.
+        Args:
+            transaction (EvmTransaction): The transaction object containing details of the swap.
+            decoded_events (list[EvmEvent]): A list of decoded events related to the transaction.
+            all_logs (list[EvmTxReceiptLog]): A list of all logs from the transaction receipt.
+        Returns:
+            list[EvmEvent]: The list of decoded events after processing the swap.
+        The function handles different types of swaps:
+        - Native token to EVM token swap, looking for the output asset is
+        eth and input asset is token
+        - EVM token to native token swap, looking for the output asset is
+        token and input asset is eth
+        - EVM token to EVM token swap, looking for input and output assets are tokens
+        It attempts to find the necessary logs for wrapping and unwrapping tokens
+        and decodes the swap accordingly.
+        If the required logs are missing, it logs an error and returns
+        the original decoded events.
+        """
+        sender = transaction.from_address
+        in_event, out_event = self._get_in_out_events(decoded_events, sender)
+        if in_event is None or out_event is None:
+            log.error(
+                f'Failed to post decoding rainbow {transaction.tx_hash.hex()} '
+                f'due to missing in_event or out_event.',
+            )
+            return decoded_events
+
+        out_asset_is_evm_token = out_event.asset.is_evm_token()
+        in_asset_is_evm_token = in_event.asset.is_evm_token()
+
+        if out_event.asset == self.evm_inquirer.native_token and in_asset_is_evm_token:
+            # Try to find if have wrap eth into weth log
+            if (wrap_event := self._find_event_by_log(all_logs, WETH_DEPOSIT_TOPIC)) is None:
+                log.error(
+                    f'Failed to decode fee from swap involving native token to token '
+                    f'for transaction {transaction.tx_hash.hex()} due to missing wrap log.',
```

Comment:
```suggestion
                    f'for transaction {transaction} due to missing wrap log.',
```

check it but the string version of the transaction includes the hash and important the chain too